examples/s3gateway/README.md

```diff
@@ -0,0 +1,152 @@
+      for file in files:
+        newpath = s3_path + dirpath + "/" +  file
+        print("copying {} to {}".format(dirpath + "/" + file, newpath))
+        file_io.copy(dirpath + "/" + file, newpath, True)
+
+    for dirpath, dirs, files in file_io.walk(s3_path +  args.inputpath, True):
+        for file in files:
+            newpath = dirpath + "/" + file
+            print("printing {} in {}  as string: >>{}<<".format(file, dirpath, file_io.read_file_to_string(newpath, False)))
+```
+
+If you want to test this in your own Kubeflow/Pachyderm deployment, 
+you can either build your own container using the `Dockerfile.tf_job_s3_gateway.yml` included with this example or
+use the prebuilt container with this code and sample data hosted in the Pachyderm docker registry.
+You create a TFJob using a TFJob manifest.
+We've included one with this example, 
+`tf_job_s3_gateway.yaml`, 
+which you would use to deploy the TFJob to your Kubeflow app
+after you've edited it for your environment.
+
+Here are the steps you need to take to run this in your environment using our prebuilt images.
+
+1. Deploy Pachyderm and Kubeflow.
+2. Edit the `S3_ENDPOINT` environment variable in `tf_job_s3_gateway.yaml` to reflect your deployment.
+   The default value,
+   `pachd.pachyderm:600`, 
+   is for Pachyderm deployed into the namespace `pachyderm` with the default S3 Gateway port.
+   If, 
+   for example, 
+   you deployed Pachyderm into the default namespace,
+   you would configure it as `pachd.default.600`, 
+   or you can hardcode it using the CLUSTER-IP address of the `pachd` service obtained through `kubectl get service -n <namespace>`.
+3. Edit the `namespace` metadata  in `tf_job_s3_gateway.yaml` to reflect your Kubeflow deployment's namespace.
+4. Create a repo in Pachyderm matching the name of the bucket used as the parameter to the `-b` argument in `tf_job_s3_gateway.yaml`.
+   If you're using the default, the command would be `pachctl create repo testrepo`.
+5. Create a branch in that repo matching the name of the bucket used as the parameter to the `-b` argument in `tf_job_s3_gateway.yaml`.
+   If you're using the default, the command would be `pachctl create branch testrepo@master`.
+   You can also just commit a file to the `repo@branch` you want to use.
+6. Deploy the manifest to your Kubeflow using  `kubectl apply -f tf_job_s3_gateway.yaml`
+7. Using your Kubeflow TFJob dashboard or `kubectl`, monitor the TFJob pod created until it completes.
+   The pod will be named after the name of the TFJob in the manifest,
+   which is `s3-gateway-example` by default, 
+   and would be something like `s3-gateway-example-worker-0`.
+8. Using your Kubeflow TFJob dashboard or `kubectl`, look at the logs for that TFJob.
+   A sample log is included in the file `sample_tf_job_logs.txt` in this example.
+
+This demonstrates that anywhere that the `file_io` library is used, 
+you can put data into and get data out of Pachyderm versioned data repositories.
```

Comment:
Ah, so I think I just figured out a major confusion of mine. You're actually adding data to pachyderm the TFJOb script first, then reading it back out? Is that right?

If so, we should reverse those as it makes a much more straightforward demo. The whole point, I think, is that users have data in pachyderm and they want to process that data with a tfjob. So they should create the repo, add the data, and then read that in tfjob, do something with it (which might be do nothing) and then write it back to pachyderm in a different repo. 

Also, we should check with eng about how provenance info would or would not get tracked here? Don't need to cover this in this example, but we should understand that.

Reply:
See above.  I'm trying to keep this as simple as possible, so folks can run this as a test to see if their setup works., then look at the code to see how to use it themselves.  Copying to Pachyderm first allows everything to be self-contained and minimized.